hibernate-core/src/main/java/org/hibernate/engine/spi/ActionQueue.java

```diff
@@ -1300,11 +1303,9 @@ private void addParentChildEntityNameByPropertyAndValue(AbstractEntityInsertActi
 					if ( !batchIdentifier.getEntityName().equals( entityName ) ) {
 						batchIdentifier.getParentEntityNames().add( entityName );
 					}
-					if ( value != null ) {
-						String valueClass = value.getClass().getName();
-						if ( !valueClass.equals( entityName ) ) {
-							batchIdentifier.getParentEntityNames().add( valueClass );
-						}
+					String valueClass = value.getClass().getName();
+					if ( !valueClass.equals( entityName ) ) {
+						batchIdentifier.getParentEntityNames().add( valueClass );
```

Comment:
Now we know `value` cannot be null

Reply:
Some tests seem to disagree :)
See `FumTest`